### DISCUSSION TITLE
NixOS

### DISCUSSION THREAD

**Discussion by agluck91** [folke/snacks.nvim#216] [Q&A] [answered]
Hello. I use NixOS and Mac as my two main machines. In both I use nix package manager to configure my systems and install vim plugins. 

There is a nix package for snacks but I have not been able to get it to work. I have been working on it for a solid day. I really love this plugin. 

I am not using lazy and based on what I saw in the code was snacks has a setup function where you can pass an opt object. It does not seem to matter what I do it is not bringing up the sub modules like notifier or dashboard. There are no errors and I have validated that it is loading the lua file where it is configured. When I try to run Snacks in the command box there is no editor commands for it at all. 

So in my lua file I run require("snacks).setup({opts})

Can some one help me figure what I am doing wrong. Are these sub modules other plugins that should be installed. Or is this all in one.

**Comment by folke** [comment]
Run `:checkhealth snacks`.

Doing `setup` is indeed what you need to do.

**Comment by agluck91** [comment]
ERROR No healthcheck found for "snacks" plugin.

I have tried both the full config on the repo, and then this trimmed down one. I also added dashboard and it did not show. Health checks always say no plugin found. And Snacks is not an editor command. 

```
require("snacks").setup({
	notifier = {
		enabled = true,
		timeout = 3000, -- default timeout in ms
	},
})
```

And here is my implementation in my neovim home manager modules. 
```
withConfig = with pkgs.vimPlugins; [
        {
          plugin = snacks-nvim;
          config = lib.fileContents ../../config/nvim/plugins/snacks.lua;
        }
...
```
I have about 20 other plugins loaded including the amazing noice!

**Reply by folke** [reply]
Then you don't have snacks installed...

**Reply by agluck91** [reply]
I am not sure why it is not installing is the question though. I am adding it like every other plugin I have added. 

Is each submodule its own plugin as well? Does Snacks do some magic under the hood?

**Reply by agluck91** [reply]
Maybe I need to message the nix maintainer for snacks. I looked at the nix packages repo for unstable and it looked right with the right commit hash.

**Comment by agluck91** [comment, marked as answer]
This functionality has been added to nixvim. I will just be switching to that.
